kiota_authentication_azure/azure_identity_access_token_provider.py

```diff
@@ -62,13 +63,15 @@ async def get_authorization_token(
                 return ""
 
             parsed_url = urlparse(uri)
-            if not all([parsed_url.scheme, parsed_url.netloc]):
+            if not all(
+                [parsed_url.scheme, parsed_url.netloc]
+            ) and parsed_url.scheme not in self.LOCALHOST_STRINGS:
                 span.set_attribute(self.IS_VALID_URL, False)
-                exc = HTTPError("Only https scheme with a valid host are supported")
+                exc = HTTPError("Valid url scheme and host required")
                 span.record_exception(exc)
                 raise exc
 
-            if not parsed_url.scheme == "https":
+            if parsed_url.scheme != "https" and (parsed_url.hostname not in self.LOCALHOST_STRINGS):
```

Comment:
is the hostname always lowercased?

Reply:
Yes, all urls are lowercased by the parsing lib.